analyzers/src/SonarAnalyzer.Common/Rules/MethodOverloadsShouldBeGroupedBase.cs

```diff
@@ -160,46 +168,37 @@ symbol switch
 
         protected class MemberInfo
         {
-
-            public readonly TMemberDeclarationSyntax Member;
-            public readonly string Accessibility;
-            public readonly SyntaxToken NameSyntax;
-            public readonly bool IsStatic;
-            public readonly bool IsAbstract;
-
+            private readonly string accessibility;
+            private readonly bool isStatic;
+            private readonly bool isAbstract;
             private readonly bool isCaseSensitive;
 
+            public TMemberDeclarationSyntax Member { get; }
+            public SyntaxToken NameSyntax { get; }
+
             public MemberInfo(SyntaxNodeAnalysisContext context, TMemberDeclarationSyntax member, SyntaxToken nameSyntax, bool isStatic, bool isAbstract, bool isCaseSensitive)
             {
                 Member = member;
-                Accessibility = context.SemanticModel.GetDeclaredSymbol(member)?.DeclaredAccessibility.ToString();
+                accessibility = context.SemanticModel.GetDeclaredSymbol(member)?.DeclaredAccessibility.ToString();
                 NameSyntax = nameSyntax;
-                IsStatic = isStatic;
-                IsAbstract = isAbstract;
+                this.isStatic = isStatic;
+                this.isAbstract = isAbstract;
                 this.isCaseSensitive = isCaseSensitive;
             }
 
-            public bool NameEquals(MemberInfo other)
-            {
-                return NameSyntax.ValueText.Equals(other?.NameSyntax.ValueText, isCaseSensitive ? StringComparison.InvariantCulture : StringComparison.InvariantCultureIgnoreCase);
-            }
+            public bool NameEquals(MemberInfo other) =>
+                NameSyntax.ValueText.Equals(other?.NameSyntax.ValueText, isCaseSensitive ? StringComparison.InvariantCulture : StringComparison.InvariantCultureIgnoreCase);
 
-            public override bool Equals(object obj)
-            {
+            public override bool Equals(object obj) =>
                 // Groups that should be together are defined by accessibility, abstract, static and member name #4136
-                return obj is MemberInfo other
+                obj is MemberInfo other
```

Comment:
I see that there are 2 uncovered branches. I assume one is when `obj is MemberInfo other` is `false`. Is the other the `accessibility` check? We might be able to easily cover it.

Reply:
The other uncovered path is `NameEquals(other) == false`, as NameEquals is already called before on that path. Yet I would not remove that line of code from the equals as it could create some confusion.